typings/index.d.ts

```diff
@@ -0,0 +1,37 @@
+// TypeScript Version: 2.8
+import {
+  SelectorMatcherOptions,
+  Matcher,
+  MatcherOptions,
+  getByTestId,
+} from 'dom-testing-library'
+
+export namespace Cypress {
+  type GetByAttribute = (id: Matcher, options?: MatcherOptions) => Chainable
```

Comment:
I've attempted to copy the implementation here: https://github.com/kentcdodds/dom-testing-library/blob/master/typings/queries.d.ts